tests/rados/test_bz_1829646.py

```diff
@@ -0,0 +1,230 @@
+    """
+    Executes du -sch to get the size of the mon DB present at /var/lib/ceph/mon/ceph-$(hostname -s)/store.db
+    :param mon_node: name of the monitor node (ceph.ceph.CephNode): ceph node
+    :return: the DB size in int
+    """
+
+    cmd = f"sudo du -ch /var/lib/ceph/mon/ceph-{mon_node.hostname}"
+    log.info(f"Collecting the size of the DB on node: {mon_node.hostname} by executing the command : {cmd}")
+    out, err = mon_node.exec_command(cmd=cmd)
+    output = out.read().decode()
+    regex = r'\s*([\d]*)[M|G]\s+[\w\W]*store.db'
+    match = re.search(regex, output)
+    size = match.groups()[0] if match else exit('could not collect the size of DB')
+    log.debug(f"the size of the cluster DB is {int(size)}")
+    return int(size)
+
+
+def get_status_from_ceph_report(mon_node, operation=None):
+    """
+    Executes command ceph report to fetch the status of the ceph cluster and collects
+    :param mon_node: name of the monitor node (ceph.ceph.CephNode): ceph node
+    :param operation: the type of info to be collected from the report.
+            operations can be : "osdmap" -> collects "osdmap_first_committed" and "osdmap_last_committed": data
+                                "health" -> collects the health status
+                                If no operation is sent, returns the entire report of the cluster
+    :return: dictionary with the requested values
+    """
+
+    cmd = r"sudo ceph report -f=json-pretty"
+    log.info(f"Collecting the status of the cluster from node: {mon_node.hostname} by executing the command : {cmd}")
+    out, err = mon_node.exec_command(cmd=cmd)
+    output = out.read().decode()
+    status_json = json.loads(output)
+    if operation.lower() == 'osdmap':
+        status_dict = {
+            "osdmap_first_committed": status_json["osdmap_first_committed"],
+            "osdmap_last_committed": status_json["osdmap_last_committed"],
+        }
+        log.debug(f"the OSD maps collected are : {str(status_dict)} ")
+    elif operation.lower() == 'health':
+        status_dict = status_json['health']
+        log.debug(f"the health status of the cluster is : {str(status_dict)} ")
+    else:
+        status_dict = status_json
+        log.debug(f"Cluster report : {str(status_dict)} ")
+    return status_dict
+
+
+def collect_osd_daemon_info(mon_node, osd_node):
+    """
+    The method is used to collect the various OSD's present on a particular node
+    :param mon_node: name of the monitor node (ceph.ceph.CephNode): ceph node
+    :param osd_node: name of the OSD node on which osd daemon details are collected (ceph.ceph.CephNode): ceph node
+    :return: list od OSD's present on the node
+    """
+
+    cmd = f"sudo ceph osd ls-tree {osd_node.hostname}"
+    log.info(f"Collecting the OSD details from node {mon_node.hostname} by executing the command : {cmd}")
+    out, err = mon_node.exec_command(cmd=cmd)
+    return [int(ids) for ids in out.read().decode().split()]
```

Comment:
can we move this to rados_utils.if possible..looks like it'll be helpful to everyone.Even i had to use this ls-tree in one of the bz

Reply:
#163